geode-core/src/main/java/org/apache/geode/internal/util/ArgumentRedactor.java

```diff
@@ -32,19 +32,68 @@
       ArrayUtils.asList(DistributionConfig.SYS_PROP_NAME, DistributionConfig.SSL_SYSTEM_PROPS_NAME,
           ConfigurationProperties.SECURITY_PREFIX);
 
-  // This pattern consists of three capture groups:
-  // The option, consisting of
-  // (a) A leading space or starting string boundary, followed by one or two hyphens
-  // (b) one or more non-whitespace, non-"=" characters, matching greedily
-  // The option-value separator, consisting of: any amount of whitespace surrounding at most 1 "="
-  // The value, consisting of:
-  // (a) If not wrapped in quotes, all non-whitespace characters, matching greedily.
-  // (b) If wrapped in quotes, any non-quote character, matching greedily, until the closing quote.
-  // -- -- This will therefore break on, e.g., --opt="escaped \" quote" and only redact "escaped."
-  // Positive lookahead between groups 1 and 2 to require space or "=", while * and ? match empty.
-  // Negative lookahead between groups 2 and 3 to avoid "--boolFlag --newOption" matching as a pair.
-  private static final Pattern optionWithValuePattern =
-      Pattern.compile("([^ ]--?[^\\s=]+)(?=[ =])( *=? *)(?!-)((?:\"[^\"]*\"|\\S+))");
+  private static final Pattern optionWithValuePattern = getOptionWithValuePattern();
+
+
+  /**
+   * This method returns the {@link java.util.regex.Pattern} given below. For clarity, the regex
+   * is given here without the escape characters required by Java's string handling.
+   * <p>
+   *
+   * ((?:^| )(?:--J=)?--?)([^\s=]+)(?=[ =])( *[ =] *)(?! *-)((?:"[^"]*"|[^"]\S+))
+   *
+   * <p>
+   * This pattern consists of one captured boundary,
+   * three additional capture groups, and two look-ahead boundaries.
+   *
+   * <p>
+   * The four capture groups are:
+   * <ul>
+   * <li>[1] The beginning boundary, including at most one leading space,
+   * possibly including "--J=", and including the option's leading "-" or "--"</li>
+   * <li>[2] The option name, which cannot include spaces</li>
+   * <li>[3] The option name / value separator, consisting of at least one character
+   * made of spaces and/or at most one "="</li>
+   * <li>[4] The option value, which terminates at the next space unless it is encapsulated by
+   * quotation-marks, in which case it terminates at the next quotation mark.</li>
+   * </ul>
```

Comment:
Add note to javadoc that the value is captured (group 4) but is not used. Capturing the value is useful for debugging.